### PR TITLE
gitlint: override requirements

### LINKED ISSUE #128998: gitlint: Broken Package
**Describe the bug**
Gitlint fails to build due to a dependency on click v7.1.2

**To Reproduce**
1. `nix-shell -p gitlint`

**Expected behavior**
Successful build of gitlint

**Screenshots**
image

**Additional context**
Looks like click got separated into two packages, one where it updated to 8.0.1 and is present in `python3Packages` and one where it was kept back for `python2Packages`.

**Notify maintainers**
@ethancedwards8 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.16.3-microsoft-standard-WSL2, Ubuntu, 20.04.2 LTS (Focal Fossa)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210601_5985b8b`
 - channels(david): `"home-manager, nixpkgs-21.11pre299378.db6e089456c"`
 - nixpkgs: `/home/david/.nix-defexpr/channels/nixpkgs`
 - 

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: gitlint
```

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Upstream is pinning their requirements and this causes build issues. 

Fixes #128998

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 129169` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitlint</li>
  </ul>
</details>

**Comment by polygon** [comment]
Since there is a derivation for Click 7.1.2, would it make sense to use this one? I've managed to do that by changing all-packages.nix like so:

```
gitlint = python3Packages.callPackage ../tools/misc/gitlint {
  click = python3Packages.callPackage ../development/python-modules/click/7.nix { };
};
```

**Comment by ethancedwards8** [comment]
> Since there is a derivation for Click 7.1.2, would it make sense to use this one? I've managed to do that by changing all-packages.nix like so:
> 
> ```
> gitlint = python3Packages.callPackage ../tools/misc/gitlint {
>   click = python3Packages.callPackage ../development/python-modules/click/7.nix { };
> };
> ```

I believe this would be the correct way to do it.

**Comment by SuperSandro2000** [comment]
> Since there is a derivation for Click 7.1.2, would it make sense to use this one? I've managed to do that by changing all-packages.nix like so:
> 
> ```
> gitlint = python3Packages.callPackage ../tools/misc/gitlint {
>   click = python3Packages.callPackage ../development/python-modules/click/7.nix { };
> };
> ```

overwriteAttr is the preferred way for applications and we try very hard to not have multiple versions of pythonPackages because mixing them causes issues.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please use
```
python3.override {
  packageOverrides = ...
}
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 129169` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitlint</li>
  </ul>
</details>

**Comment by ethancedwards8** [comment]
Hey @fabaff feel free to add yourself as a maintainer. You're clearly more knowledgeable (and active) than me, I would've had no clear where to start fixing this.

File: pkgs/tools/misc/gitlint/default.nix [hunk 1]
```diff
@@ -35,10 +66,10 @@ buildPythonApplication rec {
     gitMinimal
   ];
 
-  meta = {
+  meta = with lib; {
     description = "Linting for your git commit messages";
     homepage = "http://jorisroovers.github.io/gitlint";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/gitlint/default.nix:71]:
```suggestion
    homepage = "https://jorisroovers.com/gitlint/";
```

Comment (fabaff) [reply pkgs/tools/misc/gitlint/default.nix]: Updated, thanks.

**Pull request merged**